src/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/Pkcs/SubjectIdentifier.cs

```diff
@@ -97,6 +97,40 @@ internal SubjectIdentifier(SignerIdentifierAsn signerIdentifierAsn)
 
         public SubjectIdentifierType Type { get; }
         public object Value { get; }
+
+        public bool MatchesCertificate(X509Certificate2 certificate)
+        {
+            switch (Type)
+            {
+                case SubjectIdentifierType.IssuerAndSerialNumber:
+                    {
+                        X509IssuerSerial issuerSerial = (X509IssuerSerial)Value;
+                        byte[] serialNumber = issuerSerial.SerialNumber.ToSerialBytes();
+                        string issuer = issuerSerial.IssuerName;
+                        byte[] certSerialNumber = certificate.GetSerialNumber();
+                        if (Helpers.AreByteArraysEqual(certSerialNumber, serialNumber) && certificate.Issuer == issuer)
+                            return true;
+                    }
+                    break;
+
+                case SubjectIdentifierType.SubjectKeyIdentifier:
+                    {
+                        string skiString = (string)Value;
+                        byte[] ski = skiString.ToSkiBytes();
+                        byte[] candidateSki = PkcsPal.Instance.GetSubjectKeyIdentifier(certificate);
+                        if (Helpers.AreByteArraysEqual(ski, candidateSki))
+                            return true;
+                    }
+                    break;
+
+                default:
+                    // RecipientInfo's can only be created by this package so if this an invalid type, it's the package's fault.
+                    Debug.Fail($"Invalid recipientIdentifier type: {Type}");
```

Comment:
This type can be from a RecipientInfo, but also a SignerInfo.  I think that the comment is just out of date from a move.